### DISCUSSION TITLE
Lesson 9: Deploying your first contract ( Couldn't find injected provider )

### DISCUSSION THREAD

**Discussion by ljb630** [Cyfrin/foundry-full-course-cu#1543] [Q&A] [answered]
**Error message:**  No injected provider found. Make sure your provider (e.g. MetaMask, ...) is active and running (when recently activated you may have to reload the page).

Please check the screenshot below. I am trying to deploy my first contract and I am looking for the Environment - injected provider MetaMask but i couldn't find so, In this case how can I deploy the contract?

<img width="274" alt="Screenshot 2024-03-28 at 3 39 33 PM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/74303145/9d8ebb4e-588d-4b61-9b4b-0e787cc8b9e3">

**Comment by ljb630** [comment]
FYI, When I select injected provider its give this error

**Error message:** No injected provider found. Make sure your provider (e.g. MetaMask, ...) is active and running (when recently activated you may have to reload the page).

**Reply by EngrPips** [reply]
hello @ljb630 , do you have metmask installed on the browser where you are currently trying to use remix ?

**Reply by ljb630** [reply]
@EngrPips Yes i already installed it and now am using it as a chrome extension.

**Reply by EngrPips** [reply]
Hello @ljb630 , please use the `WalletConnect` option if you are having problem with uisng the `injected Provider`  option. a picture below show the `WalletConnect` option. 

<img width="371" alt="Screenshot 2024-03-28 at 6 46 00 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/c9131bbf-14f6-49b1-824b-cf6711d68c54">

That should work fine. I am looking forward to hear back from you.

**Reply by ljb630** [reply]
Ok let me try this then!

**Reply by EngrPips** [reply]
Alright, looking forward to hear how it turns out

**Reply by ljb630** [reply]
@EngrPips Finally, I was unable to deploy my Remix IDE on my PC, but I was able to do so using Web Remix.

Etherscan link: https://sepolia.etherscan.io/tx/0x3a378547e6f6b66b2d22507627c652a8334ba280a599f0bebecb1e0d047092b3

<img width="1029" alt="Screenshot 2024-03-28 at 11 13 23 PM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/74303145/0e002722-c170-43b0-93e3-c029bfa0fa98">

**Reply by EngrPips** [reply]
Awesome keep crushing it buddy, glad you was finally able to deploy your contract

**Comment by ljb630** [comment, marked as answer]
@EngrPips Thank you so much, my man you are so useful :)

**Reply by EngrPips** [reply]
Anytime buddy, I am very glad i am of use to someone at least.

**Reply by MrGbengaAdams** [reply]
this just helped me through lesson 4

**Reply by EngrPips** [reply]
I am glad it helped you man. Keep crushing it!

**Reply by muhammadnasir-py** [reply]
thanks Man, this helped me as well. Very grateful.

**Reply by EngrPips** [reply]
Kepp crushing it!
